image/src/routes/ipfs.ts

```diff
@@ -0,0 +1,151 @@
+import { Hono } from 'hono'
+import { cors } from 'hono/cors'
+import { CACHE_DAY, Env } from '../utils/constants'
+import { fetchIPFS } from '../utils/ipfs'
+import { ipfsToCFI } from '../utils/cloudflare-images'
+import { allowedOrigin } from '../utils/cors'
+import type { CFIApiResponse } from '../utils/types'
+
+const app = new Hono<{ Bindings: Env }>()
+
+app.use('/*', cors({ origin: allowedOrigin }))
+app.get('/*', async (c) => {
+  const { original } = c.req.query()
+  const isOriginal = original === 'true'
+
+  const url = new URL(c.req.url)
+  const path = url.pathname.replace('/ipfs/', '')
+  const fullPath = `${path}${url.search}`
+
+  // TODO: check response from cache
+  // ----------------------------------------
+  let response = undefined
+  console.log('response', response)
+
+  // contruct r2 object
+  const objectName = `ipfs/${path}`
+  const object = await c.env.MY_BUCKET.get(objectName)
+  const mimeType = object?.httpMetadata?.contentType
+  console.log('object', object)
+  console.log('mime type', mimeType)
+
+  // 1. check existing image on cf-images && !isOriginal
+  // ----------------------------------------
+  console.log('step 1')
+  if (mimeType?.includes('image') && !isOriginal) {
+    const getImage = await fetch(
+      `https://api.cloudflare.com/client/v4/accounts/${c.env.CF_IMAGE_ACCOUNT}/images/v1/${fullPath}`,
+      {
+        method: 'GET',
+        headers: {
+          'Content-Type': 'application/json',
+          Authorization: `Bearer ${c.env.IMAGE_API_TOKEN}`,
+        },
+      }
+    )
+    const image = (await getImage.json()) as CFIApiResponse
+
+    if (getImage.ok && image.result) {
+      const variants = image.result.variants
+      const publicURL = variants.find((url) => url.endsWith('/public'))
+
+      if (publicURL) {
+        return c.redirect(publicURL)
+      }
+    }
+  }
```

Comment:
Move to utils/cloudflare image so this can be 3 loc.

```ts
const publicURL = await getImageByPath(path) 

if (publicURL) {
   return c.redirect(publicURL)
 }
```